front/src/modules/simulationResult/components/SpeedSpaceChart/SpeedSpaceChart.tsx

```diff
@@ -165,6 +166,23 @@ export default function SpeedSpaceChart({
     }
   }, [resetChart]);
 
+  /**
+   * Recompute the enabled axes when the settings change
+   */
+  useEffect(() => {
+    const settingToAxisMap = {
+      altitude: CHART_AXES.SPACE_HEIGHT,
+      curves: CHART_AXES.SPACE_RADIUS,
+      slopes: CHART_AXES.SPACE_GRADIENT,
+    };
+    const newEnabledAxes = Object.entries(localSettings)
+      .filter(
+        ([key, isEnabled]) => isEnabled && settingToAxisMap[key as keyof typeof settingToAxisMap]
+      )
+      .map(([key]) => settingToAxisMap[key as keyof typeof settingToAxisMap]);
+    setEnabledAxes(newEnabledAxes.map((tuple) => [...tuple]));
+  }, [localSettings]);
```

Comment:
I think it's better to use a useMemo that is recomputed each time localSettings changes. Also, you can move the const settingToAxisMap outside the component to prevent if from being reinstantiated at each re-render (and also rename it SETTING_TO_AXIS in capital letters)
```suggestion
  const additionalCurves = useMemo(
    () =>
      Object.keys(settingToAxisMap)
        .filter((key) => localSettings[key as keyof typeof localSettings])
        .map((key) => settingToAxisMap[key as keyof typeof settingToAxisMap] as ChartAxes),
    [localSettings]
  );
```

Reply:
Okay, but this array is of type ChartAxes, so I still prefer to call it additionalAxes